pkg/apis/sources/v1alpha1/gitlabsource_lifecyle.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2020 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"sort"
+	"strings"
+)
+
+// String prepended to GitLab event types to make them fully-qualified.
+const eventPrefixGitLab = "dev.knative.sources.gitlab."
+
+// Types of events emitted by a GitLabSource.
+// The chosen format an case matches the "object_kind" attribute contained in
+// payloads sent by GitLab's webhooks.
+// https://docs.gitlab.com/ee/user/project/integrations/webhooks.html#events
+const (
+	GitLabEventTypeBuild        = "build"
+	GitLabEventTypeDeployment   = "deployment"
+	GitLabEventTypeIssue        = "issue"
+	GitLabEventTypeMergeRequest = "merge_request"
+	GitLabEventTypeNote         = "note"
+	GitLabEventTypePipeline     = "pipeline"
+	GitLabEventTypePush         = "push"
+	GitLabEventTypeTagPush      = "tag_push"
+	GitLabEventTypeWikiPage     = "wiki_page"
+)
+
+// Types of webhooks that can be enabled on a GitLab project.
+// https://docs.gitlab.com/ee/api/projects.html#add-project-hook
+const (
```

Comment:
I have not checked if this covers all webhooks possible but I will trust you

Reply:
I appreciate the trust